spec/turn_spec.rb

```diff
@@ -156,6 +156,49 @@
     end
   end
 
+  describe "#check_diagonal_win" do 
+    it "checks for horizontal wins" do 
+      player = Player.new
+      board = Board.new
+      turn = Turn.new(player, board)
+      
+      turn.check_diagonal_win
+      
+      expect(player.winner).to be false
+      
+      board.board_grid[0][0].set_state("x")
+      board.board_grid[1][1].set_state("x")
+      board.board_grid[2][2].set_state("x")
+      board.board_grid[3][3].set_state("x")
+      
+      turn.check_diagonal_win
+      
+      expect(player.winner).to be true
+    end
+  end
+  
+  describe "#diag_offset_columns" do 
+    it "returns an array of all diagonals from columns 1 - 3" do 
+      player = Player.new
+      board = Board.new
+      turn = Turn.new(player, board)
+    
+      expect(turn.diag_offset_columns.size).to eq(3)
```

Comment:
So these are a bit hard-coded for sure, because WE know the board size, but if the project didn't include a static board it could be problematic.